packages/zero-tag-processor/src/utils/cssFunctionTransformerPlugin.ts

```diff
@@ -82,16 +76,19 @@ const cssFunctionTransformerPlugin = declare<BabelPluginOptions>((api, pluginOpt
         if (themeKey === 'typography' && val === 'inherit') {
           return;
         }
+        const propertyThemeKey = themeKey ?? val.split('.')[0];
```

Comment:
Found one more bug. With:

```js
export const Main = styled('main')({
  …,
  '--test': (props) => props.test,
});
```

produces:

<img width="230" alt="image" src="https://github.com/mui/material-ui/assets/18292247/5f5217ca-0304-4c14-b5e7-f41854b70122">

I think the variable should not be created if the result of the callback is `undefined | null | ''`

Reply:
I tried to fix it but I am not sure where to find.

Reply:
It'll need to be fixed in styled.ts of zero-runtime package since this function is called at runtime.
Right now it only checks for number/string. It'll need to account for undefined

Reply:
Fixed in latest commit.